ui/src/app/base/hooks/base.test.tsx

```diff
@@ -218,4 +218,13 @@ describe("hooks", () => {
       expect(result.current).toBe(false);
     });
   });
+
+  describe("getId", () => {
+    it("generates the id on first render", () => {
+      const { result, rerender } = renderHook(() => useId());
+      const previousResult = result;
+      rerender();
+      expect(result).toEqual(previousResult);
```

Comment:
@petermakowski `renderHook` has an API similar to refs where you need to access the current result so I've done a drive-by fix in my PR: https://github.com/canonical-web-and-design/maas-ui/pull/3577/files#diff-b41dc6d828d01e96b1cd53babeadcac324d8de4c2a38e9deddaa8c98826db028L223.

I've also added an initial check that the ID is set otherwise this test could pass with `expect(undefined).toEqual(undefined);`